presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -112,4 +123,47 @@ public static boolean isEqualComparisonExpression(Expression expression)
     {
         return expression instanceof ComparisonExpression && ((ComparisonExpression) expression).getOperator() == ComparisonExpression.Operator.EQUAL;
     }
+
+    static Optional<EnumType> maybeGetEnumTypeFromLiteral(QualifiedName qualifiedName, TypeManager typeManager)
+    {
+        if (qualifiedName.getParts().size() < 2) {
+            // an enum literal should be of the form `MyEnum.my_key`
+            return Optional.empty();
+        }
+        try {
+            Type baseType = typeManager.getType(parseTypeSignature(qualifiedName.getParts().get(0)));
```

Comment:
It's the second to the last part that you are looking for. I'd do `qualifiedName.getPrefix().getSuffix()` here. Or get the size of `get(size - 2)`.